src/ESLint.Formatter/sarif.js

```diff
@@ -175,7 +191,10 @@ module.exports = function (results, data) {
                                 };
                                 if (meta.docs.description) {
                                     sarifRules[message.ruleId].shortDescription = {
-                                        text: meta.docs.description
+                                        text: formatRuleText(meta.docs.description)
+                                    };
+                                    sarifRules[message.ruleId].help = {
```

Comment:
The purpose of setting help property here is to show the rule description in github code scanning alert page, without it the page only renders tool name and rule id. 

There is only a description field in ESLint's rule meta data (example below). 

Should we keep only help and remove shortDescription property?

            "no-undef": {
                "type": "problem",
                "docs": {
                    "description": "disallow the use of undeclared variables unless mentioned in `/*global */` comments",
                    "recommended": true,
                    "url": "[https://eslint.org/docs/rules/no-undef](https://eslint.org/docs/rules/no-undef)"
                },
                "schema": [
                    {
                        "type": "object",
                        "properties": {
                            "typeof": {
                                "type": "boolean",
                                "default": false
                            }
                        },
                        "additionalProperties": false
                    }
                ],
                "messages": {
                    "undef": "'{{name}}' is not defined."
                }
            },

Reply:
I would check in the github documentation what is the required field.

Reply:
checked the github [doc](https://docs.github.com/en/code-security/code-scanning/integrating-with-code-scanning/sarif-support-for-code-scanning) it seems out-dated.
I tested in a test repo and see the actual fields, please see the screenshot [here](https://user-images.githubusercontent.com/76094515/153503208-3636e24c-aa57-4ddd-bb9c-f5c7eca8aa08.png): 
- Code scanning alert title: Sarif rule short description
- Code scanning alert description under title: Sarif rule full description
- Code scanning alert issue description inline: Sarif rule help

Looks like we need set at lease short description (title) and help text (description in inline). What do you think?

Reply:
added another fix for rule.shortDescription property: convert markdown to plain text, as an issue mentioned by customer the ESLint code scanning alert title shows raw markdown text.

Reply:
Why did you make this decision? It seems like you are reducing the richness of SARIF in order to work around a bug in another system. 

Do we have a GitHub specific validator for this problem (if I'm understanding it correctly)? Did someone add a converter update to make the correct transformation in our SDK? Did you reach out to GH on the issue?

Can you please get a working example together of how your ESLint alerts are getting rendered in the GH UI so that everyone can help you test/think about this?